src/passes/DeadArgumentElimination.cpp

```diff
@@ -281,7 +294,7 @@ struct DAE : public Pass {
       auto name = pair.first;
       // We can only optimize if we see all the calls and can modify
       // them.
-      if (infoMap[name].hasUnseenCalls) {
+      if (infoMap[name].hasUnseenCalls.load()) {
```

Comment:
I believe you can imply on implicit conversion operators instead of explicitly calling `load` in nearly all cases. This would be more readable.

```suggestion
      if (infoMap[name].hasUnseenCalls) {
```